### ISSUE TITLE
How could i set the mount option like noresvport nolock

### ISSUE THREAD

**Issue by lefu999** [sahlberg/libnfs#237]
As i know ,i can set mount -o noresvport,nolock,hard,intr,tcp,timeo=60,vers=3...etc -t nfs  /mnt/name in linux,i can`t found how to set the option in this project.Please help me,Thx!!!!!

**Comment by sahlberg** [comment]
libnfs does not support either of those options.

Reserved ports:
When an application using libnfs is running as root, or has cap_net_bind_service set, then libnfs will use a privileged port. Otherwise libnfs will use a non-privileged port.

Locking:
libnfs supports both lockf() and fcntl() locking with NFSv4.
There is no way to disable the locking support.
If the application does not want/need locking then the app just have to not call the locking functions in the libnfs api.

**Comment by sahlberg** [comment]
For fuse-nfs:
fuse-nfs does not yet support any kind of locking so the behavior with fuse will be similar to what nolock provides.

**Comment by lefu999** [comment]
@sahlberg thx... when i use libnfs instead of the linux kernel nfs in the same environment , i found it slower than  before ,would you give me some advice?

**Comment by sahlberg** [comment]
On Tue, Jan 30, 2018 at 12:43 PM, lefu999 <notifications@github.com> wrote:

> @sahlberg <https://github.com/sahlberg> thx... when i use libnfs instead
> of the linux kernel nfs in the same environment , i found it slower than
> before ,would you give me some advice?
>

fuse-nfs + libnfs will always be a little,sometimes a it more, slower than
the kernel nfs driver for almost all workloads.

There are several reasons for this. For example, latency will suffer
because of the extra context switches it takes to bounce between the kernel
and user-land fuse
module. In case of fuse-nfs there will also be some extra copying of memory
between buffers that is not necessary in the kernel module.

But the biggest reason is probably that the kernel module can, and do, do a
lot of read and write caching. Something that is not really feasible to do
safely in userspace.


So basically, it is expected to be somewhat slower. You can try using
fuse-nfs with NFSv4.
This is very new support and not fully tested  but due to the nature of the
NFSv4 protocol, there are certain things
that libnfs can do to speed things up compared to the default version which
uses NFSv3.


See the readme file on how to try NFSv4.

**Comment by lefu999** [comment]
@sahlberg thx,i try to use nfsv4 it show "Failed to mount nfs share : mount_cb: Bad digit in fattr3 uid",and i found thd *buff(uid or gid) is "nobody".i try to append&uid=xxx&gid=xxx to the nfs url or call rpc_set_uid or nfs_set_uid and it did`t work too..
in nfsv3:fuse-nfs -n nfs:/xxx -m /mnt/test2 ,it is okay
in nfsv4:fuse-nfs -n nfs:/xxx -m /mnt/test2?version=4 ,it fail...

**Comment by sahlberg** [comment]
Can you send me a wireshark trace of when this happens?
It sounds like a bug.

**Comment by lefu999** [comment]
@sahlberg I send the wireshark trace to your gmail.

**Comment by lefu999** [comment]
[fuse-nfsv4.log](https://github.com/sahlberg/libnfs/files/1685380/fuse-nfsv4.log)
@sahlberg upload to here

**Comment by lefu999** [comment]
@sahlberg 
i test between the libnfs and the kernel nfs write/read speed and i found the kernel faster  triple than the lib nfs.
Write performance:
1024+0 records in
1024+0 records out
8388608 bytes transferred in 0.031 secs (270600258 bytes/sec)
    0m0.09s real     0m0.02s user     0m0.06s system
root@rk3288:/ # time dd if=/mnt/test3/test of=/dev/null bs=8k                  
1024+0 records in
1024+0 records out
8388608 bytes transferred in 1.126 secs (7449918 bytes/sec)
    0m1.18s real     0m0.05s user     0m0.03s system

Read performance:
1024+0 records in
1024+0 records out
8388608 bytes transferred in 2.530 secs (3315655 bytes/sec)
    0m2.55s real     0m0.02s user     0m0.09s system

1024+0 records in
1024+0 records out
8388608 bytes transferred in 0.818 secs (10255022 bytes/sec)
    0m0.87s real     0m0.04s user     0m0.07s system

**Comment by sahlberg** [comment]
Hi,

The problem is/was that the server returned the string 'nobody' instead of
a pure numerical value.
I have changed it so that IF the function getpwnam() is available on your
platform, then it will try to
use that function to convert the uid/gid string that the server returned
into a uid/gid value.

If that fails too I will now set the uid/gid as 65534 which is legacy
values that are used for nobody in nfs.


Please try current libnfs master.

**Comment by lefu999** [comment]
@sahlberg 
i have merged your commit and it mount for nfsv4 successfully but when i use "ls-l" it show lstat './xxx' failed: No such file or directory and i see the tcp dump it show "Status: NFS4ERR_NOENT (2)".It worked normally for nfsv3.

**Comment by lefu999** [comment]
@sahlberg  hi,anyupdate?
